src/field.cpp

```diff
@@ -2579,10 +2579,16 @@ bool field_type_dangerous( field_id id )
     return ft.dangerous[0] || ft.dangerous[1] || ft.dangerous[2];
 }
 
-void map::emit_field( const tripoint &pos, const emit_id &src )
+void map::emit_field( const tripoint &pos, const emit_id &src, float mul )
```

Comment:
Not sure how to do it otherwise.
`emit` sources are very un-universal because there is literally no way to scale them other than creating an array and picking from it.